akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala

```diff
@@ -390,5 +394,14 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
     testActor.asInstanceOf[ActorRef[AnyRef]] ! Stop
   }
 
+  def tell(m: M) = testActor.tell(m)
+
+  // impl InternalRecipientRef
+  def provider: ActorRefProvider =
+    ActorSystemAdapter(system.classicSystem).asInstanceOf[ActorSystemAdapter[Nothing]].provider
```

Comment:
ActorSystemAdapter is to adapt a classic system to a typed, but we already have a typed system and want to access the provider, so `system.classicSystem.asInstanceOf[ExtendedActorSystem].provider` would make more sense IMO

Reply:
I wasn't completely sure that the `classicSystem` would always be an `ExtendedActorSystem`, but if that's something which can reasonably be presumed to hold in the future...

Reply:
Yep, that assumption is made all over the place so it won't change any time soon.